concrete-core/src/backends/core/private/crypto/ggsw/ciphertext.rs

```diff
@@ -432,4 +462,220 @@ impl<Cont> GgswCiphertext<Cont> {
                 )
             })
     }
+
+    pub fn fill_with_trivial_encryption<Scalar>(&mut self, plaintext: &Plaintext<Scalar>)
+    where
+        Self: AsMutTensor<Element = Scalar>,
+        Scalar: UnsignedTorus,
+    {
+        // We fill the ggsw with trivial glwe encryptions of zero:
+        for mut glwe in self.as_mut_glwe_list().ciphertext_iter_mut() {
+            let mut mask = glwe.get_mut_mask();
+            mask.as_mut_tensor().fill_with_element(Scalar::ZERO);
+        }
+        let base_log = self.decomposition_base_log();
+        for mut matrix in self.level_matrix_iter_mut() {
+            let decomposition = plaintext.0.wrapping_mul(
+                Scalar::ONE
+                    << (<Scalar as Numeric>::BITS
+                        - (base_log.0 * (matrix.decomposition_level().0))),
+            );
+            // We iterate over the rows of the level matrix
```

Comment:
I am not sure we have what is needed in `backends::core::private::math::decomposition`, but it would be nice to either use it, or move this snippet there, and call it here. The same goes for the external product actually. What do you think @agnesLeroy ?

Reply:
I don't know about the external product but for this one why not :)

Reply:
@orelcosseron we need to put the code to create `decomposition` somewhere to use it throughout ciphertext files. I will open an issue so we can keep it in the backlog and do that later if it's ok for you @aPere3.